spec/inputs/redis_spec.rb

```diff
@@ -61,3 +72,183 @@ def process(conf, event_count)
     process(conf, event_count)
   end
 end
+
+# unit tests ---------------------
+
+describe Logstash::Inputs::Redis do
+  let(:redis) { double('redis') }
+  let(:builder) { ->{ redis } }
+  let(:connection) { double('redis_connection') }
+  let(:connected) { [true] }
+  let(:data_type) { 'list' }
+  let(:cfg) { {'key' => 'foo', 'data_type' => data_type} }
+  let(:quit_calls) { [:quit] }
+
+  subject do
+    described_class.new(cfg).add_external_redis_builder(builder)
+  end
+
+  context 'construction' do
+    it 'registers the input' do
+      expect {subject.register}.not_to raise_error
+    end
+  end
+
+
+  context 'runtime for list data_type' do
+    before do
+      subject.register
+    end
+
+    context 'teardown when redis is unset' do
+      let(:quit_calls) { [:quit, :unsubscribe, :punsubscribe, :connection, :disconnect!] }
+
+      it 'does not attempt to quit' do
+        allow(redis).to receive(:nil?).and_return(true)
+        quit_calls.each do |call|
+          expect(redis).not_to receive(call)
+        end
+        expect {subject.teardown}.not_to raise_error
+      end
+    end
+
+    let(:accumulator) { [] }
```

Comment:
I'm not sure this needs to be a `let`, seems like you can just do `accumulator = []` in the spec? Or maybe it's a `let` for DRY purposes?